src/client/Microsoft.Identity.Client/Internal/Requests/SilentBrokerAuthStretegy.cs

```diff
@@ -1,48 +1,59 @@
 using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Cache;
+using Microsoft.Identity.Client.Cache.Items;
 using Microsoft.Identity.Client.Core;
-using Microsoft.Identity.Client.Internal.Requests;
+using Microsoft.Identity.Client.Internal.Broker;
 using Microsoft.Identity.Client.OAuth2;
 using Microsoft.Identity.Client.Utils;
 
-namespace Microsoft.Identity.Client.Internal.Broker
+namespace Microsoft.Identity.Client.Internal.Requests
 {
-    internal class BrokerSilentRequest
+    internal class SilentBrokerAuthStretegy : ISilentAuthStrategy
     {
-        public Dictionary<string, string> BrokerPayload
-            = new Dictionary<string, string>();
-        internal IBroker Broker { get; }
+        internal AuthenticationRequestParameters _authenticationRequestParameters;
+        private ICacheSessionManager CacheManager => _authenticationRequestParameters.CacheSessionManager;
+        protected IServiceBundle _serviceBundle;
         private readonly AcquireTokenSilentParameters _silentParameters;
-        private readonly AuthenticationRequestParameters _authenticationRequestParameters;
-        private readonly IServiceBundle _serviceBundle;
-        private readonly ICoreLogger _logger;
+        private SilentRequest _silentRequest;
+        internal IBroker Broker { get; }
+        public Dictionary<string, string> BrokerPayload = new Dictionary<string, string>();
+        ICoreLogger _logger;
 
-        internal BrokerSilentRequest(
-            AuthenticationRequestParameters authenticationRequestParameters,
-            AcquireTokenSilentParameters acquireTokenSilentParameters,
+        public SilentBrokerAuthStretegy(
+            SilentRequest request,
             IServiceBundle serviceBundle,
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenSilentParameters silentParameters,
             IBroker broker)
         {
             _authenticationRequestParameters = authenticationRequestParameters;
-            _silentParameters = acquireTokenSilentParameters;
+            _silentParameters = silentParameters;
             _serviceBundle = serviceBundle;
+            _silentRequest = request;
             Broker = broker;
-            _logger = _authenticationRequestParameters.RequestContext.Logger;
+            _logger = authenticationRequestParameters.RequestContext.Logger;
+        }
+
+        public async Task<AuthenticationResult> ExecuteAsync(CancellationToken cancellationToken)
+        {
+            var response = await SendTokenRequestToBrokerAsync().ConfigureAwait(false);
+            return await _silentRequest.CacheTokenResponseAndCreateAuthenticationResultAsync(response).ConfigureAwait(false);
         }
 
         public async Task<MsalTokenResponse> SendTokenRequestToBrokerAsync()
         {
-            if(!Broker.IsBrokerInstalledAndInvokable())
+            if (!Broker.IsBrokerInstalledAndInvokable())
```

Comment:
there won't be a null ref here right? with `Broker.`

Reply:
It's passed in the ctor, but we don't check for nulls ...

Reply:
right, that's what I meant. Are we checking for null anywhere before we get this far.